### PR TITLE
Add FHIR IG importer to platform

### PR DESCRIPTION
**Pull request by drizzentic**
Add FHIR IG Upload mediator and microfrontend UI to platform

### CODE REVIEW

**Comment by rcrichton** [comment]
Task linked: [CU-86byatwb7 Add FHIR IG Upload app to platform](https://app.clickup.com/t/86byatwb7)

File: config.yaml [hunk 1]
```diff
@@ -73,3 +74,10 @@ profiles:
       - openhim-mapping-mediator
     envFiles:
       - mpi.env
+
+  - name: ig-importer
+    packages:
+      - fhir-ig-importer
+    envFiles:
+      - cdr.env
+    dev: true
```
Comment (bradsawadye) [inline config.yaml:83]: this is not necessary. We should only have recipes' profiles

File: fhir-ig-importer/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    networks:
+      ig-importer:
+      hapi-fhir:
+      openhim:
+    deploy:
+      replicas: ${FHIR_IG_IMPORTER_UI_INSTANCES}
+      placement:
+        max_replicas_per_node: 1
+    environment:
+      FHIR_IG_IMPORTER_CORE_URL: ${FHIR_IG_IMPORTER_CORE_URL}
```
Comment (bradsawadye) [inline fhir-ig-importer/docker-compose.yml:31]:
```suggestion
      FHIR_IG_IMPORTER_CORE_URL: ${FHIR_IG_IMPORTER_CORE_URL}

```

File: fhir-ig-importer/package-metadata.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    "fhir-datastore-hapi-fhir"
+  ],
+  "environmentVariables": {
+    "HAPI_FHIR_BASE_URL": "http://hapi-fhir:8080/fhir",
+    "HAPI_FHIR_INSTANCES": 1,
+    "FHIR_IG_IMPORTER_CORE_PORT": 3001,
+    "FHIR_IG_IMPORTER_CORE_HOST": "0.0.0.0",
+    "FHIR_IG_IMPORTER_CORE_URL": "http://0.0.0.0:3001/fhir/ig/v1.0",
+    "FHIR_IG_IMPORTER_UI_VERSION": "latest",
+    "FHIR_IG_IMPORTER_CORE_VERSION": "latest",
```
Comment (bradsawadye) [inline fhir-ig-importer/package-metadata.json:18]: We should pin these to specific versions

File: interoperability-layer-openhim/docker-compose.yml [hunk 4]
```diff
@@ -2,7 +2,7 @@ version: "3.9"
 
 services:
   openhim-core:
-    image: jembi/openhim-core:v8.4.0
+    image: jembi/openhim-core:microfrontends-pre-release
```
Comment (bradsawadye) [inline interoperability-layer-openhim/docker-compose.yml:5]: Use an env variable to specify the version

**Comment by bradsawadye** [comment]
@drizzentic looks good. Left some minor comments

**Review by BMartinos — COMMENTED** [review]
overall looks good to me, just a couple of comments/questions

File: fhir-ig-importer/docker-compose.dev.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+services:
+  fhir-ig-importer-mediator:
+    ports:
+      - target: 3001
+        published: 3333
+        mode: host
+  fhir-ig-importer-ui:
+    ports:
+      - target: 8080
```
[the remark this answers is not in the document]
Comment (drizzentic) [reply fhir-ig-importer/docker-compose.dev.yml:12]: @BMartinos This is for a dev setup and we can always change it. But I will give it a port that is not being used.

File: fhir-ig-importer/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+services:
+  fhir-ig-importer-mediator:
+    image: jembi/fhir-ig-importer-mediator:${FHIR_IG_IMPORTER_CORE_VERSION}
+    networks:
+      ig-importer:
+      hapi-fhir:
+      openhim:
+    deploy:
+      replicas: ${FHIR_IG_IMPORTER_CORE_INSTANCES}
```
[the remark this answers is not in the document]
Comment (drizzentic) [reply fhir-ig-importer/docker-compose.yml:11]: @BMartinos This won't be needed. We will only have one instance as per @bradsawadye suggestion.

**Review by BMartinos — APPROVED** [review]
Just two minor comments, but not blocking the PR from being merged

File: interoperability-layer-openhim/package-metadata.json [hunk 7]
```diff
@@ -6,6 +6,7 @@
   "type": "infrastructure",
   "dependencies": [],
   "environmentVariables": {
+    "OPENHIM_CORE_IMAGE": "jembi/openhim-core:microfrontends-pre-release",
```
[the remark this answers is not in the document]
Comment (drizzentic) [reply interoperability-layer-openhim/package-metadata.json:9]: Hi @BMartinos,
Yes, the new image is needed and is based off the poc-microfrontends branch in the console repo. The console stores the base/core apps needed but the FHIR IG imported is more of an extension which wasn't bundled together but build and bundled as independent app which comes with it's mediator and UI.

**Review by bradsawadye — APPROVED** [review]

**Pull request merged**